src/main/java/com/celements/pagetype/service/PageTypeService.java

```diff
@@ -57,6 +62,48 @@
   @Requirement
   private IModelAccessFacade modelAccess;
 
+  @Requirement
+  private List<IPageTypeCategoryRole> pageTypeCategoryList;
+
+  private final ConcurrentMap<String, IPageTypeCategoryRole> typeNameToCategoryMap = new ConcurrentHashMap<>();
+
+  private ConcurrentMap<String, IPageTypeCategoryRole> getTypeNameToCategoryMap() {
+    if (typeNameToCategoryMap.isEmpty()) {
+      synchronized (typeNameToCategoryMap) {
+        if (typeNameToCategoryMap.isEmpty()) {
+          for (IPageTypeCategoryRole typeCategory : pageTypeCategoryList) {
+            IPageTypeCategoryRole beforeRegCat = typeNameToCategoryMap.putIfAbsent(
+                typeCategory.getTypeName(), typeCategory);
```

Comment:
perhaps better use com.google.common.collect.ImmutableMap.Builder instead of using and returning a modifiable ConcurrentHashMap.